src/track/external-thumbnails-handler.js

```diff
@@ -0,0 +1,263 @@
+import Player from '../player';
+import EventManager from '../event/event-manager';
+import FakeEventTarget from '../event/fake-event-target';
+import getLogger from '../utils/logger';
+import {Parser, StringDecoder} from './text-track-display';
+import * as Utils from '../utils/util';
+import {ThumbnailInfo} from '../thumbnail/thumbnail-info';
+import {Cue} from './vtt-cue';
+
+const DEFAULT_THUMBNAIL_SIZE: {width: number, height: number} = {
+  width: 164,
+};
+
+const VTT_WITH_IMG_SIZE: RegExp = /#xy=/i;
+const VTT_WITH_IMG_SIZE_AND_COORDS: RegExp = /#xywh=/i;
+
+class ExternalThumbnailsHandler extends FakeEventTarget {
+  /**
+   * constructor
+   * @param {Player} player - the player instance.
+   */
+  constructor(player: Player) {
+    super();
+    this._player = player;
+    this._eventManager = new EventManager();
+  }
+
+  /**
+   * The external thumbnails handler class logger.
+   * @type {any}
+   * @static
+   * @private
+   */
+  static _logger: any = getLogger('ExternalThumbnailsHandler');
+
+  /**
+   * the player instance.
+   * @type {Player}
+   * @private
+   */
+  _player: Player;
+
+  /**
+   * event manager for the external thumbnails handler
+   * @type {EventManager}
+   * @private
+   */
+  _eventManager: EventManager;
+
+  /**
+   * the processed thumbnail cues
+   * @type {Array<Cue>}
+   * @private
+   */
+  _cues: Array<PKThumbnailVttCue> = [];
+
+  /**
+   * base url of the images or the sprite img
+   * @type {string}
+   * @private
+   */
+  _imgBaseUrl: string;
+
+  /**
+   * computed img dimensions based on its natural ratio
+   * @type {Object}
+   * @private
+   */
+  _computedImgSize: {width:number, height: number} = {width: null, height: null};
+
+  /**
+   * start the loading and parsing process of the vtt thumbnails file.
+   * @returns {void}
+   * @public
+   */
+  async load(): Promise<void> {
+    const thumbnailsConfig: PKExternalThumbnailsConfig = this._player.sources.thumbnails;
+    if (!thumbnailsConfig) {
+      return [];
```

Comment:
return

Reply:
the method signature returns a promise

Reply:
and it dose return promise cause it async function